extensions/src/main/java/org/apache/falcon/extensions/mirroring/hdfsSnapshot/HdfsSnapshotMirroringExtension.java

```diff
@@ -0,0 +1,239 @@
+        }
+
+        String tdeEnabled = extensionProperties.getProperty(
+                HdfsSnapshotMirrorProperties.TDE_ENCRYPTION_ENABLED.getName());
+        if (StringUtils.isBlank(tdeEnabled)) {
+            additionalProperties.put(HdfsSnapshotMirrorProperties.TDE_ENCRYPTION_ENABLED.getName(), "false");
+        }
+
+        // Add sourceCluster properties
+        Cluster sourceCluster = ClusterHelper.getCluster(extensionProperties.getProperty(
+                HdfsSnapshotMirrorProperties.SOURCE_CLUSTER.getName()));
+        if (sourceCluster == null) {
+            LOG.error("Cluster entity {} not found", HdfsSnapshotMirrorProperties.SOURCE_CLUSTER.getName());
+            throw new FalconException("Cluster entity "
+                    + HdfsSnapshotMirrorProperties.SOURCE_CLUSTER.getName() + " not found");
+        }
+        additionalProperties.put(HdfsSnapshotMirrorProperties.SOURCE_NN.getName(),
+                ClusterHelper.getStorageUrl(sourceCluster));
+        additionalProperties.put(HdfsSnapshotMirrorProperties.SOURCE_EXEC_URL.getName(),
+                ClusterHelper.getMREndPoint(sourceCluster));
+        String sourceKerberosPrincipal = ClusterHelper.getPropertyValue(sourceCluster, SecurityUtil.NN_PRINCIPAL);
+        if (StringUtils.isNotBlank(sourceKerberosPrincipal)) {
+            additionalProperties.put(HdfsSnapshotMirrorProperties.SOURCE_NN_KERBEROS_PRINCIPAL.getName(),
+                    sourceKerberosPrincipal);
+        }
+
+        String sourceRetentionPolicy = extensionProperties.getProperty(
+                HdfsSnapshotMirrorProperties.SOURCE_SNAPSHOT_RETENTION_POLICY.getName());
+        if (StringUtils.isBlank(sourceRetentionPolicy)) {
+            sourceRetentionPolicy = DEFAULT_RETENTION_POLICY;
+        }
+        validateRetentionPolicy(sourceRetentionPolicy);
+        additionalProperties.put(HdfsSnapshotMirrorProperties.SOURCE_SNAPSHOT_RETENTION_POLICY.getName(),
+                sourceRetentionPolicy);
+
+        // Add targetCluster properties
+        Cluster targetCluster = ClusterHelper.getCluster(extensionProperties.getProperty(
+                HdfsSnapshotMirrorProperties.TARGET_CLUSTER.getName()));
+        if (targetCluster == null) {
+            LOG.error("Cluster entity {} not found", HdfsSnapshotMirrorProperties.TARGET_CLUSTER.getName());
+            throw new FalconException("Cluster entity "
+                    + HdfsSnapshotMirrorProperties.TARGET_CLUSTER.getName() + " not found");
+        }
+        additionalProperties.put(HdfsSnapshotMirrorProperties.TARGET_NN.getName(),
+                ClusterHelper.getStorageUrl(targetCluster));
+        additionalProperties.put(HdfsSnapshotMirrorProperties.TARGET_EXEC_URL.getName(),
+                ClusterHelper.getMREndPoint(targetCluster));
+        String targetKerberosPrincipal = ClusterHelper.getPropertyValue(targetCluster, SecurityUtil.NN_PRINCIPAL);
+        if (StringUtils.isNotBlank(targetKerberosPrincipal)) {
```

Comment:
There are issues with respect to configuration (with ipc fallback to simple auth enabled) and there are situations where the generated workflows should work independent of the cluster where they run is secure or not.   OOZIE-2356 allows the ability to skip credentials at action level depending on the cluster is secure or not which will be needed to fully realize this